language/move-core/types/src/effects.rs

```diff
@@ -0,0 +1,46 @@
+use crate::{
+    account_address::AccountAddress,
+    identifier::Identifier,
+    language_storage::{StructTag, TypeTag},
+};
+use anyhow::Result;
+use std::collections::btree_map::{self, BTreeMap};
+
+/// A collection of changes to modules and resources under a Move account.
+#[derive(Clone)]
+pub struct AccountChangeSet {
+    pub modules: BTreeMap<Identifier, Option<Vec<u8>>>,
```

Comment:
It might be slightly clearer to use a `Write(Vec<u8>), Delete` enum here instead of `Option`.